Drakhtar/Board.cpp

```diff
@@ -1,21 +1,24 @@
 #include "Board.h"
 
-Board::Board(Texture* cf, int r, int c, int cs) : GameObject(cf, (800 - (cs * c)) / 2, (600 - (cs * r)) / 2, cs * c, cs * r) {
+Board::Board(Texture* cellTexture, int r, int c, float cellSize) : GameObject(nullptr, 0, 0, 0, 0) {
 	rows = r;
 	cols = c;
-	cellSize = cs;
-	cellFrame = cf;
+
+	// Calcula los m�rgenes horizontales y verticales
+	float marginX = (800 - (cellSize * (cols - 1))) / 2;
+	float marginY = (600 - (cellSize * (rows - 1))) / 2;
 
 	// Crea el tablero
-	board = new GameObject**[rows];
+	board = new Box**[rows];
 	for (int i = 0; i < rows; i++) {
-		board[i] = new GameObject*[cols];
+		board[i] = new Box*[cols];
 	}
 
-	// Rellena el tablero de nullptrs
+	// Rellena el tablero de cajas vac�as
 	for (int i = 0; i < rows; i++) {
 		for (int j = 0; j < cols; j++) {
-			board[i][j] = nullptr;
+			vector2d pos = { marginX + j * cellSize, marginY + i * cellSize };
+			board[i][j] = new Box(cellTexture, pos, { cellSize, cellSize }, nullptr);
```

Comment:
For consistency and readability, I would do this, but other than that it's fine:
```suggestion
			vector2d size = { cellSize, cellSize };
			board[i][j] = new Box(cellTexture, pos, size, nullptr);
```